src/metabase/api/user_key_value.clj

```diff
@@ -0,0 +1,35 @@
+(ns metabase.api.user-key-value
```

Comment:
General feedback
1. **More RESTful endpoints** - I'm seeing in the notion doc that “Reading other’s users data” is a non-goal, but I think an endpoint format like `api/user/:userId/setting/:key` would be a bit more RESTful and follow our settings endpoints a bit more. (If we do this `api/user/current/setting/:key` would be a nice ergonomic way to adjust just the current user's setting.) This could also come in handy by letting admins change user settings for other users if we ever needed.
2. **Bulk endpoint** - I think we can quickly eat up the browser's ~6 max in-flight requests to a domain, so having a way for the FE to bulk fetch could avoid introducing latency to other requests on the page. Following on the format above, maybe something like `GET /api/user/:userId/setting?keys=aaa,bbb,ccc`?

Reply:
for 2 I'd vote to use the fact that querystring is ok with multiple keys and doing `/setting?key=aaa&key=bbb&key=ccc` (for John: we're doing something similar with `ms/QueryVectorOf` in serialization API).

for 1... IDK, I'd rather not have user in the url, and if we every need user id overrides - query parameters are always there.

Reply:
for 1: 

> I think an endpoint format like api/user/:userId/setting/:key would be a bit more RESTful and follow our settings endpoints a bit more

I see how it's more restful, but I don't think having the user-id in the route params would be very helpful in practice. We already get the user-id on every authenticated request, so it'll always be handy. Also since a user can only ever set/retrieve their own kvs, the `/user/8/` piece of the url would always be identical, and require another check for `user-id-from-route` == `api/*current-user-id*`.

If we ever decide to let admins change user settings for other users this seems like the right kind of route to add, tho